asset/style.scss

```diff
@@ -592,6 +638,113 @@ footer {
     background: $warm-5;
     color: $warm-dark;
   }
+
+  // Syntax highlighting - Dark theme
+  @media (prefers-color-scheme: dark) {
+
+    pre { color: $body-color-dark; }
+
+    .k { color: #B89FF7; }                  // Keyword.
+    .n { color: #0FDEBD; }                  // Number.
+    .s { color: #16C98D; }                  // String.
+    .e { color: hsl(45, 80%, 55%); }        // String escape.
+    .c { color: #2C98EC; }                  // Comment.
+    .a { color: #B89FF7; }                  // Preprocessor, annotation.
+    .i { color: #FF708E; }                  // Identifier.
+    .t { color: #FFC83F;}                   // Type name.
```

Comment:
I'm flattered! If this were to be used for real though, I think the syntax colors should match a bit better with the light version. This uses the same palette, but rearranged to keep the same token -> color association:

```
    .k { color: #2C98EC; }                  // Keyword.
    .n { color: #FF953F; }                  // Number.
    .s { color: #FFD78B; }                  // String.
    .e { color: hsl(45, 80%, 55%); }        // String escape.
    .c { color: #79858D; }                  // Comment.
    .a { color: #B89FF7; }                  // Preprocessor, annotation.
    .i { color: #70C2FF; }                  // Identifier.
    .t { color: #0FDEBD; }                  // Type name.
```
![crafting-interpreters-dark-mode-comparison](https://user-images.githubusercontent.com/13191586/103263230-40accd80-4975-11eb-954d-7c9492397503.png)

A couple of other suggestions:
- Convert all the new colors to HSL for consistency with the rest of the code
- Massage the slate blues used for the backgrounds in the dark version to make them match the the sidebar better. They're _close_ as-is, but I suspect they need to be desaturated and maybe spun towards green by a tiny smidge. Converting them to HSL first should make the discrepancy more obvious. 🙂

Reply:
Thanks for the feedback! And the new colour tweaks.

HSL is great for _creating_ colour schemes, but has little extra value once colours have been chosen. That said, I may switch them as you suggest for general consistency. Although it's worth noting that the original Sass can be inconsistent. Eg.

`body {
  background: #382611 url('image/wood.jpg') top center / 100% auto no-repeat;
}`

Thanks for attempting to match the light scheme. I actually really like your VS Code palette though :) I feel it's generally more readable, despite its deviation from the original light scheme. I wonder what Bob thinks?

Reply:
Switched HEX values to HSL.

Reply:
Do you have new suggestions for your final point? I struggled a bit with the backgrounds for the "Challenges" and "Design Note" sections.

Reply:
> Thanks for attempting to match the light scheme. I really like your [VS Code palette](https://vscodethemes.com/e/dannymcgee.summer-vacation-theme) though :) I feel it's possibly a bit more readable, and looks great, despite its deviation from the original light scheme.

Thank you! I like it too (obviously). 🙂 My theme is pretty similar to Atom One Dark, while the Crafting Interpreters theme is similar to the default Visual Studio/VS Code dark scheme. I suspect the latter is more familiar to most developers, but since I started writing JavaScript in Atom years ago, those colors are kind of hard-wired into my brain. Either way, I agree the VS-alike palette is a better fit for the site aesthetically.

> Do you have new suggestions for your final point? I struggled a bit with the backgrounds for the "Challenges" and "Design Note" sections.
> 
> Edit: I've taken a few more stabs at it. See what you think of the new version.

I like it! I'll leave it up to Bob at this point if he wants to move forward with his own suggestions.